c7n/resources/apigw.py

```diff
@@ -406,6 +409,80 @@ class resource_type(query.TypeInfo):
         cfn_type = 'AWS::ApiGateway::VpcLink'
 
 
+@resources.register('rest-client-certificate')
+class RestClientCertificate(query.QueryResourceManager):
+    """TLS client certificates generated by API Gateway
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: old-client-certificates
+                resource: rest-client-certificate
+                filters:
+                  - key: createdDate
+                    value_type: age
+                    value: 90
+                    op: greater-than
+    """
+    class resource_type(query.TypeInfo):
+        service = 'apigateway'
+        enum_spec = ('get_client_certificates', 'items', None)
+        id = 'clientCertificateId'
+        name = 'client_certificate_id'
+        permissions_enum = ('apigateway:GET',)
+        cfn_type = 'AWS::ApiGateway::ClientCertificate'
+
+
+@RestStage.filter_registry.register('client-certificate')
+class StageClientCertificateFilter(RelatedResourceFilter):
+    """Filter API stages by a client certificate
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: rest-stages-old-certificate
+                resource: rest-stage
+                filters:
+                  - type: client-certificate
+                    key: createdDate
+                    value_type: age
+                    value: 90
+                    op: greater-than
+    """
+    schema = type_schema('client-certificate', rinherit=ValueFilter.schema)
+    RelatedResource = "c7n.resources.apigw.RestClientCertificate"
+    annotation_key = "c7n:matched-client-certificate"
+    source_related_id_map = {
+        "config": "ClientCertificateId",
```

Comment:
feels like we need a resource type metadata field for resources that use lower case keys, to avoid the de-mangling we normally do from config.

Reply:
added in #6377 to handle this more generically, but it will require walking back through extant usages to verify

Reply:
I was wondering about a more generic way to handle the related ID expression, didn't think bigger than that though. Good idea.